std/typecons.d

```diff
@@ -2944,6 +2959,19 @@ auto nullable(T)(T t)
     assert(ntts.to!string() == "2.5");
 }
 
+// Bugzilla 14477
+@safe unittest
+{
+    static struct DisabledDefaultConstructor
+    {
+        @disable this();
+        this(int i) { }
+    }
+    Nullable!DisabledDefaultConstructor var;
+    var = DisabledDefaultConstructor(5);
```

Comment:
Save the value and do an assert here.

Reply:
What do you mean?

Reply:
Sorry, the idea was to assert that the  custom constructor has been called

Reply:
Seems a bit redundant - the custom constructor is just so I can make a value of that type at all, and the point of the unittest is to test Nullable, not @disable this or struct initialization.